lib/internal/Magento/Framework/App/Bootstrap.php

```diff
@@ -441,6 +441,10 @@ protected function terminate(\Throwable $e)
             } catch (\Exception $e) {
                 $message .= "Could not write error message to log. Please use developer mode to see the message.\n";
             }
+            if (!headers_sent()) {
+                header('HTTP/1.1 503 Service Temporarily Unavailable');
```

Comment:
1. I believe it should be 500 error, right?
2. In case if that's developer mode I think we need to send this status header also.